packages/abi-coder/src/encoding/coders/OptionCoder.ts

```diff
@@ -25,7 +25,7 @@ export class OptionCoder<TCoders extends Record<string, Coder>> extends EnumCode
   }
 
   private toOption(output?: DecodedValueOf<TCoders>): Option<unknown> {
-    if (output && 'Some' in output) {
+    if (output && Object.hasOwn(output, 'Some')) {
```

Comment:
I believe that the following check is causing the issues.
https://github.com/FuelLabs/fuels-ts/blob/45f3ef2bac2999f876e1390f0670ac7df5737f8b/packages/abi-coder/src/encoding/coders/EnumCoder.ts#L45-L47

We've changed constant value for `OPTION_CODER_TYPE` to `enum std::option::Option` (in this [PR](https://github.com/FuelLabs/fuels-ts/pull/2856)).

https://github.com/FuelLabs/fuels-ts/blob/45f3ef2bac2999f876e1390f0670ac7df5737f8b/packages/abi-coder/src/utils/constants.ts#L14

The [type](https://github.com/FuelLabs/fuels-ts/blob/929ac2d33d53254d2e7fa4201955686f4e17d540/packages/abi-coder/src/encoding/coders/AbstractCoder.ts#L34) of the coder has remained the same (`enum Option`) as we're only pulling off the [last name](https://github.com/FuelLabs/fuels-ts/blob/45f3ef2bac2999f876e1390f0670ac7df5737f8b/packages/abi-coder/src/encoding/strategies/getCoderV1.ts#L130-L150) from the ABI type. This leads to the `isNativeEnum` method now returns `true` rather than `false` in the instance of an `Option`. This then leads to `None` being returned as the decoded value, rather than the correct result of `{ None: undefined }`.

---

We could change the [`lastNameRegExMatch`](https://github.com/FuelLabs/fuels-ts/blob/45f3ef2bac2999f876e1390f0670ac7df5737f8b/packages/abi-coder/src/utils/constants.ts#L29) to obtain the full namespace type so for case of an Option, it would get `std::option::Option`.

```ts
export const lastNameRegExMatch = /([^\s]+)$/m;
```

We could also make two constants:

```ts
const `RAW_ABI_OPTION_TYPE` = `enum std::option::Option`; // Can be used in the ABI
const `OPTION_CODER_TYPE` = `enum Option`;
```

Reply:
Got it. You can try it out and see how it goes.

We can circle back later.

Reply:
Thanks for the fix via:
 - https://github.com/FuelLabs/fuels-ts/pull/2870